test/openAPI/features/support/g2p_update_beneficiary_details.js

```diff
@@ -0,0 +1,188 @@
+const chai = require('chai');
+const { spec } = require('pactum');
+const { Given, When, Then, Before, After } = require('@cucumber/cucumber');
+const {
+  localhost,
+  defaultExpectedResponseTime,
+  contentTypeHeader,
+  acceptHeader,
+  updateBeneficiaryDetailsEndpoint,
+  g2pResponseSchema,
+} = require('./helpers/helpers');
+
+chai.use(require('chai-json-schema'));
+
+const baseUrl = localhost + updateBeneficiaryDetailsEndpoint;
+const endpointTag = { tags: `@endpoint=/${updateBeneficiaryDetailsEndpoint}` };
+
+Before(endpointTag, () => {
+  specUpdateBeneficiaryDetails = spec();
+});
+
+// Scenario: Succesfully updated beneficiary details smoke type test
+Given(
+  /^Information Mediator BB wants to update beneficiary details$/,
+  () => 'Information Mediator BB wants to update beneficiary details'
+);
+
+When(
+  /^POST request to update beneficiary details with given "([^"]*)" as RequestID, "([^"]*)" as SourceBBID and "([^"]*)" as PayeeFunctionalID is sent$/,
+  (requestID, sourceBBID, payeeFunctionalID) =>
+    specUpdateBeneficiaryDetails
+      .post(baseUrl)
+      .withHeaders(acceptHeader.key, acceptHeader.value)
+      .withJson({
+        RequestID: requestID,
+        SourceBBID: sourceBBID,
+        Beneficiaries: [{ PayeeFunctionalID: payeeFunctionalID }],
+      })
+);
+
+Then(
+  /^The response from the \/update\-beneficiary\-details is received$/,
+  async () => await specUpdateBeneficiaryDetails.toss()
+);
+
+Then(
+  /^The \/update\-beneficiary\-details response should be returned in a timely manner 15000ms$/,
+  () =>
+    specUpdateBeneficiaryDetails
+      .response()
+      .to.have.responseTimeLessThan(defaultExpectedResponseTime)
+);
+
+Then(
+  /^The \/update\-beneficiary\-details response should have status (\d+)$/,
+  status => {
+    specUpdateBeneficiaryDetails.response().to.have.status(status);
+  }
+);
+
+Then(
+  /^The \/update\-beneficiary\-details response should have content\-type: application\/json header$/,
+  () =>
+    specUpdateBeneficiaryDetails
+      .response()
+      .should.have.header(contentTypeHeader.key, contentTypeHeader.value)
+);
+
+Then(
+  /^The \/update\-beneficiary\-details response should match json schema$/,
+  () =>
+    chai
+      .expect(specUpdateBeneficiaryDetails._response.json)
+      .to.be.jsonSchema(g2pResponseSchema)
+);
+
+// Scenario: Succesfully updated beneficiary details using not obligatory fields in the request payload
+// Others Given, When, Then are written in the aforementioned example
+When(
+  /^POST request to update beneficiary details with given "([^"]*)" as RequestID, "([^"]*)" as SourceBBID, "([^"]*)" as PayeeFunctionalID, "([^"]*)" as PaymentModality and "([^"]*)" as FinancialAddress is sent$/,
+  (
+    requestID,
+    sourceBBID,
+    payeeFunctionalID,
+    paymentModality,
+    financialAddress
+  ) =>
+    specUpdateBeneficiaryDetails
+      .post(baseUrl)
+      .withHeaders(acceptHeader.key, acceptHeader.value)
+      .withJson({
```

Comment:
`.withBody` may be more intuitive

Reply:
I added .withJson because it seems that .withBody prefers string values like ( string(json), string(xml), files without using form-data ). 

wBody: https://pactumjs.github.io/api/requests/withBody.html
wJson: https://pactumjs.github.io/api/requests/withJson.html

Reply:
Okay